decomp/semantics/uds.py

```diff
@@ -974,19 +1023,14 @@ def sentence_id(self) -> str:
         return self.syntax_nodes[syntax_root]['sentence_id']
 
 
-
-class UDSDataset:
-    """A Universal Decompositional Semantics dataset
-
-    Parameters
-    ----------
-    annotation
-        mapping from node ids or pairs of node ids separated by %% to
-        attribute-value pairs; node ids must not contain %%
+class UDSAnnotation(ABC):
```

Comment:
I've put the implementation of a number of methods in abstract methods in this class and have the implementing classes simply call `super().method_name()`. Hope that's all right.